src/sqlfluff/core/rules/std/L043.py

```diff
@@ -0,0 +1,104 @@
+"""Implementation of Rule L043."""
+
+from sqlfluff.core.rules.base import BaseCrawler, LintFix, LintResult
+from sqlfluff.core.rules.doc_decorators import document_fix_compatible
+
+
+@document_fix_compatible
+class Rule_L043(BaseCrawler):
+    """Use an explicit condition rather than case when statement.
+
+    | **Anti-pattern**
+
+    .. code-block:: sql
+
+        select
+            case
+                when fab > 0 then true else false end as is_fab
+        from fancy_table
+
+    | **Best practice**
+
+    .. code-block:: sql
+
+        select
+            fab > 0 as is_fab
+        from fancy_table
+
+    """
+
+    def _eval(self, segment, **kwargs):
+        """Find rule violations and provide fixes.
+
+        0. Look for a case expression
+        1. Find the first expression and "then"
+        2. Determine if "then" is followed by a boolean
```

Comment:
IIUC, the code checks for a Boolean after `THEN`, and if it finds one, it assumes the rest of the expression is:
```
else <<opposite value>>
```

What happens if this is not the case, e.g. if there's no else (result will be `NULL`, I think), or if it's `else <<complex expression>>>`?

In those cases, could we detect it doesn't fit the pattern and do nothing?

Reply:
Yeah that's a good callout. Will narrow this rule down to only handle: `case when... then bool else bool end`.

Along these lines, now that I think about it a little more, we might also need to wrap the fixed expression in a `coalesce`.

Will try to make these updates later today.

Reply:
Sounds great!